eng/common/scripts/stress-testing/deploy-stress-tests.ps1

```diff
@@ -1,3 +1,4 @@
+[CmdletBinding(DefaultParameterSetName = 'Default')]
```

Comment:
This shouldn't be necessary. All parameters not in a parameter set belong to __AllParameters anyway.

Reply:
I will test. I think I was having some issues when this wasn't defined with parameters in the other defined parameter set. becoming required.

Reply:
If I don't include this line, then it seems to require the other parameter set:

```
$ deploy-stress-tests.ps1 -Environment prod -CI

cmdlet deploy-stress-tests.ps1 at command pipeline position 1
Supply values for the following parameters:
Login:
```

Reply:
It's possible since it seems like there's only one other parameter set, which begs the question why one was declared in the first place. A parameter set should only be declared if there are 2 or more distinct parameter sets. The original only had 1: `DoLogin`. It should've had zero, or define a second one that makes them distinct. Any parameters not in a specific parameter set (or more: `[Parameter(...)]` can be repeated) belong to all parameter sets.

Reply:
After discussing offline, I understand the intent now is that if one of the parameters in the `DoLogin` parameter set is specified, you want both to be; thus, defining the default param set as "Default" (anything, really, other than "__AllParameters") creates a pseudo parameter set to which those two `DoLogin` parameters don't belong, while every other parameter belongs to all of them.

As such, this makes sense and no point in changing it, though you might consider a comment to this effect.

Reply:
Chatted offline, keeping the current behavior but adding a comment to make it clear why it's set up that way.